src/Reflection/AbstractFunction.php

```diff
@@ -364,64 +349,46 @@ public function __get($key)
      * @param mixed $value
      * @return void
      */
-    public function __set($key, $value)
+    public function __set(string $key, $value): void
     {
         $this->config[$key] = $value;
     }
 
     /**
      * Set method's namespace
      *
-     * @param string $namespace
      * @throws Exception\InvalidArgumentException
-     * @return void
      */
-    public function setNamespace($namespace)
+    public function setNamespace(?string $namespace = null): void
```

Comment:
probably we should not have the default value here, we want to have `setNamespace` called always with the value.
Calling setter without any value is always confusing.

Reply:
Agreed. Remove default value. Additionally set `$namespace` to `null` if empty, since this would mean an non-present state.